app/Compiler/Twig/templates/baseRecord.twig

```diff
@@ -34,14 +40,47 @@ abstract class BaseRecord implements JsonSerializable
 
             try {
                 $prop = $refl->getProperty($propertyToSet);
-            } catch (\ReflectionException $e) {
+            } catch (ReflectionException $e) {
                 continue;
             }
 
-            if ($prop instanceof ReflectionProperty) {
-                $prop->setAccessible(true);
-                $prop->setValue($this, $value);
+            if (!$prop instanceof ReflectionProperty) {
+                continue;
+            }
+
+            $propName = $prop->getName();
+            if (key_exists($propName, $this->propClassMap)) {
```

Comment:
Maybe move `protected $propClassMap = [];` above this function. I get why declaring it above `getValueForComplexPropType` has advantages too, but I was confused when I saw this the first time

Reply:
Oh yeah forgot I use it up here.